### PR TITLE
Make local solver in NLDD a runtime specified ISTLSolver

### PR DESCRIPTION
**Pull request by jakobtorben**
For the NLDD nonlinear solver, we are solving a linear system locally for each subdomain. Currently the linear solver for this task is hardcoded to use ilu0 and default max iterations and reduction tolerance. For more flexibility and easier experiments I propose we set this as a command line argument using the non-invasive changes in this PR.

### CODE REVIEW

File: opm/simulators/linalg/FlowLinearSolverParameters.cpp [hunk 1]
```diff
@@ -107,6 +122,11 @@ void FlowLinearSolverParameters::registerParameters()
          "cpr_trueimpes, cpr_trueimpesanalytic, amg or hybrid (experimental). "
          "Alternatively, you can request a configuration to be read from a "
          "JSON file by giving the filename here, ending with '.json.'");
+    Parameters::Register<Parameters::NlddLocalLinearSolver>
```
Comment (atgeirr) [inline opm/simulators/linalg/FlowLinearSolverParameters.cpp:125]: This looks good, but I realized we have forgotten to change the help message for --linear-solver above when we changed the default linear solver, can you change it to say that cprw is default while changing this file anyway?

**Review by atgeirr — COMMENTED** [review]
Looks good, will merge when minor issues are addressed.

File: opm/simulators/linalg/FlowLinearSolverParameters.cpp [hunk 2]
```diff
@@ -56,6 +54,19 @@ void FlowLinearSolverParameters::init(bool cprRequestedInDataFile)
         linsolver_ = Parameters::Get<Parameters::LinearSolver>();
     }
 
+    // if local solver from nldd, use nldd local linear solver parameters
+    if (is_nldd_local_solver_) {
+        is_nldd_local_solver_ = true;
```
[the remark this answers is not in the document]
Comment (jakobtorben) [reply opm/simulators/linalg/FlowLinearSolverParameters.cpp]: You are indeed right... From an older version

File: opm/simulators/linalg/ISTLSolver.hpp [hunk 3]
```diff
@@ -238,12 +238,20 @@ std::unique_ptr<Matrix> blockJacobiAdjacency(const Grid& grid,
...
+
+                // Do a normal linear solver setup.
+                if (parameters_[0].is_nldd_local_solver_) {
+                    prm_.push_back(setupPropertyTree(parameters_[0],
+                                                 Parameters::IsSet<Parameters::NlddLocalLinearSolverMaxIter>(),
+                                                 Parameters::IsSet<Parameters::NlddLocalLinearSolverReduction>()));
+                }
+                else {
+                    prm_.push_back(setupPropertyTree(parameters_[0],
                                                  Parameters::IsSet<Parameters::LinearSolverMaxIter>(),
```
[the remark this answers is not in the document]
Comment (jakobtorben) [reply opm/simulators/linalg/ISTLSolver.hpp]: Fixed

**Review by atgeirr — APPROVED** [review]

**Comment by atgeirr** [comment]
jenkins build this please

**Pull request merged**